iothub/service/src/RegistryManager.cs

```diff
@@ -68,161 +69,134 @@ public void Dispose()
         /// Register a new device with the system
         /// </summary>
         /// <param name="device">The Device object to be registered.</param>
-        /// <returns>echoes back the Device object with the generated keys and ETags</returns>
+        /// <returns>echoes back the Device object with the generated keys and ETags.</returns>
         public abstract Task<Device> AddDeviceAsync(Device device);
 
         /// <summary>
         /// Register a new device with the system
         /// </summary>
         /// <param name="device">The Device object to be registered.</param>
         /// <param name="cancellationToken">The token which allows the operation to be canceled.</param>
-        /// <returns>echoes back the Device object with the generated keys and ETags</returns>
+        /// <returns>echoes back the Device object with the generated keys and ETags.</returns>
         public abstract Task<Device> AddDeviceAsync(Device device, CancellationToken cancellationToken);
 
         /// <summary>
         /// Register a new module with device in the system
         /// </summary>
         /// <param name="iotHubModule">The Module object to be registered.</param>
-        /// <returns>echoes back the Module object with the generated keys and ETags</returns>
+        /// <returns>echoes back the Module object with the generated keys and ETags.</returns>
         public abstract Task<Module> AddModuleAsync(Module iotHubModule);
 
         /// <summary>
         /// Register a new module with device in the system
         /// </summary>
         /// <param name="iotHubModule">The Module object to be registered.</param>
         /// <param name="cancellationToken">The token which allows the operation to be canceled.</param>
-        /// <returns>echoes back the Module object with the generated keys and ETags</returns>
+        /// <returns>echoes back the Module object with the generated keys and ETags.</returns>
         public abstract Task<Module> AddModuleAsync(Module iotHubModule, CancellationToken cancellationToken);
 
         /// <summary>
         /// Adds a Device with Twin information
         /// </summary>
-        /// <param name="device"></param>
-        /// <param name="twin"></param>
-        /// <returns></returns>
+        /// <param name="device">The device to add.</param>
+        /// <param name="twin">The twin information for the device to be added.</param>
+        /// <returns>The result of the add operation.</returns>
         public abstract Task<BulkRegistryOperationResult> AddDeviceWithTwinAsync(Device device, Twin twin);
```

Comment:
Weird that adding a single device returns a "Bulk"RegistryOperationResult, right?

Reply:
Good point. I believe that's because even though we're adding a single device, the underlying operations puts that device in a list and does a Bulk device operation. And that returns a bulk result.